pkg/cli/app.go

```diff
@@ -59,6 +61,7 @@ type app struct {
 	RPromptPersistent func() bool
 	BeforeReadline    []func()
 	AfterReadline     []func(string)
+	AfterCommand      []func(string, float64, error)
```

Comment:
The after-command hook does not belong in the `App` type; the `App` type is only responsible for the "R" (read) part in a REPL, while an after-command hook is only of concern for the whole REPL lifecycle.

In fact, the after-command hook doesn't belong in the `Editor` type either, because the `Editor` type is also only concerned about the R part in REPL. Eventually there should be a `REPL` type that takes care of REPL lifecycle stuff and that would be an appropriate home for the after-command hook.

But before that is done, the `Editor` type is a better temporary home than `App`, so move it there.

Reply:
Done. And I'll note that moving the implementation also reduced the footprint of this change since it no longer touches *pkg/cli/app.go*. But that raises the question why this belongs in the Editor implementation but the before and after readline callbacks do not.